spiceaidocs/docs/features/local-acceleration/constraints.md

```diff
@@ -0,0 +1,144 @@
+---
+title: 'Constraints'
+sidebar_label: 'Constraints'
+sidebar_position: 2
+description: 'Learn how to add/configure constraints on local acceleration tables in Spice.'
+---
+
+Constraints are rules that enforce data integrity in a database. Spice supports constraints on locally accelerated tables to ensure data quality, as well as configuring the behavior for inserting data updates that violate constraints.
+
+Constraints are specified in the Spicepod via the `primary_key` field in the acceleration configuration. Additional unique constraints are specified via the [`indexes`](./indexes.md) field with the value `unique`.
+
+The behavior of inserting data that violates the constraint can be configured via the `on_conflict` field to either `drop` the data that violates the constraint or `upsert` that data into the accelerated table (i.e. update all values other than the columns that are part of the constraint to match the incoming data).
+
+If there are multiple rows in the incoming data that violate any constraint, the entire incoming batch of data will be dropped.
```

Comment:
Multiple is 2 or more?

Reply:
Yeah, if there are two rows that are duplicated in the incoming data - then the entire batch will be dropped.